enhancements/machine-config/machine-config-node.md

```diff
@@ -0,0 +1,340 @@
+
+The states to be reported by this MachineConfigNode will roughly fall into the following:
+
+#### Preparing phase
+ - stopping config drift monitor
+ - reconciling configs
+#### Actionable phase
+- Draining
+- Updating on disk state
+- Updating OS
+#### Post update action
+- Rebooting
+- Closing daemon
+- Node Reboot **OR** Reloading crio
+
+This loops until we are inDesiredConfig
+
+More Paths that will update the node state:
+
+#### Completing Update -- the daemon only gets into this phase (outside of hypershift) when we are inDesiredConfig,it has to do with the drainer annotation
+- Uncordoning
+- Uncordoned
+- Updating node state and metrics
+   		 
+#### Resume phase
+- Start config drift monitor
+- Error states:
+   - update stuck -- specifically degraded and/or failures in the functions this is the phase that happens if we either error in any of the above stages
+   - unavailable nodes... an obstacle has been hit, but its not the MCOs fault
+     - Disk Pressure
+   	 - Unschedulable
+
+
+### Workflow Description
+
+When an upgrade is triggered by there being mismatch between a desired and current config or simply just a new MachineConfig being applied, the MachineConfigNodes for a specific pool will report the following processes (roughly)
+
+
+```console
+$ oc get machineconfignodes
+NAME                          UPDATED   UPDATEPREPARING   UPDATEINPROGRESS   UPDATEPOSTACTION   UPDATECOMPLETING   RESUMING
+ip-10-0-12-194.ec2.internal   True      False             False              False              False              False
+ip-10-0-17-102.ec2.internal   True      False             False              False              False              False
+ip-10-0-2-232.ec2.internal    True      False             False              False              False              False
+ip-10-0-59-251.ec2.internal   True      False             False              False              False              False
+ip-10-0-59-56.ec2.internal    True      False             False              False              False              False
+ip-10-0-6-214.ec2.internal    True      False             False              False              False              False
+```
+
+```console
+$ oc get machineconfignodes
+NAME                          UPDATED   UPDATEPREPARING   UPDATEINPROGRESS   UPDATEPOSTACTION   UPDATECOMPLETING   RESUMING
+ip-10-0-12-194.ec2.internal   False     True              False              False              False              False
+ip-10-0-17-102.ec2.internal   True      False             False              False              False              False
```

Comment:
Has this node already completed its update to have a `True` in the `Updated` column here?  I'd expect a transition when the machine-config controller assigns a new target-MachineConfig to the MachineConfigPool where some "I am aimed at the pool's current target" condition  flips from `True` to `False` for all the nodes in the pool except for the `maxUnavailable` nodes selected for the first round of updates.

Reply:
Perhaps it would be useful to print `spec.configVersion.desired` as a column as well?

Reply:
This seems to be pretty narrowly scoped to "Is this node currently at its desired config, regardless of whether the pool is updating and moving individual nodes to a new desired config?"  I think I like the current `UPDATED` and `UPDATING` values, T/F indicating that the node is current with the pool config, F/F indicating that it's waiting to update to pool level, and F/T indicating that it's in progress of updating to pool level

Reply:
@sdodson if I understand what you are saying is you want all of the nodes to report `Updated`: False immediately when the new config is generated. However, nodes should only show updating when they are acting upon this mismatch and doing upgrade stuff? If that is correct, then I agree with you and will take that into account when implementing in the MCO!

Reply:
> If that is correct, then I agree with you and will take that into account when implementing in the MCO!

This works for me